### PR TITLE
logging: start logger thread in enable_logger()

### LINKED ISSUE #11008: net-related logging: Don't see the expected output + crash in logging
Context: I'm developing a new network driver. I never wrote one from scratch, so doing a lot of stupid things. I would think that logging is there to help me, but instead, it puzzles me.

So, to reproduce this one, you'd need to look at the branch https://github.com/pfalcon/zephyr/tree/smsc9220-pfalcon-logging-issue . There lives the new WIP driver, and to test it, I made adhoc prj.conf changes to hello_world and echo_server samples. The changes are largely the same and include CONFIG_ETHERNET_LOG_LEVEL_DBG=y. I run:

~~~
hello_world$ zephyr-make BOARD=mps2_an385 run
[QEMU] CPU: cortex-m3
init ok
eth_smsc9220_isr: 0 0
eth_initialize
eth_tx
eth_tx
***** Booting Zephyr OS zephyr-v1.13.0-1494-gfabf4596e1 *****
[00:00:00.000,000] <err> eth_smsc9220.eth_init: MAC 52:54:00:12:34:56
Hello World! mps2_an385
~~~

That's the expected output (specifically `<err>` logging).

Now I run:

~~~
echo_server$ zephyr-make BOARD=mps2_an385 run
[QEMU] CPU: cortex-m3


uart:~$ init ok
eth_smsc9220_isr: 0 0
eth_initialize
eth_tx
eth_tx
***** Booting Zephyr OS zephyr-v1.13.0-1494-gfabf4596e1 *****
eth_tx
~~~

Which is not the expected output, as I miss that err logging of mac address.

(If you run these samples, you'll notice that soon after that output, they crash. Well, that's the issue I'd like logging to help me with, and not sure if it can, because from my PoV it works erratically even with simpler things.)

### PR DESCRIPTION
**Pull request by mike-scott**
With this change, logging starts at the same time whether or not
CONFIG_LOG_PROCESS_THREAD is enabled/disabled.  Previously,
if disabled, the POST_KERNEL trigger would start logger much
earlier in the init allowing debugging of network drivers
which might be holding back the call to main().

With CONFIG_LOG_PROCESS_THREAD enabled, logging wouldn't start
till _init_static_threads() in bg_thread_main() which is just
before the call to main().

Fixes #11008.

Signed-off-by: Michael Scott <mike@foundries.io>

### CODE REVIEW

**Comment by mike-scott** [comment]
Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/10733

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by pfalcon** [comment]
> Previously,
if disabled, the POST_KERNEL trigger would start logger much
earlier

So, does this patch moves the logger to start "much later" instead? If yes, I wonder if it's good change. And if no, then it's worth to update the commit message to avoid questions like that.

**Comment by nordic-krch** [comment]
It is better that logger processing starts in the same phase regardless of CONFIG_LOG_PROCESS_THREAD.
Could you explain a bit what was the issue you have seen?
Logger core initialization consists of 2 stages. At first (as soon as possible, before 'anything' else ) log_core is initialized (`log_core_init`). Once it is initialized, log messages can be created and put to the queue. However, engine for processing is not yet started. Then log thread is initialized (`log_enable`). But then still no message will be processed until at least one backend is enabled (that was added by https://github.com/zephyrproject-rtos/zephyr/commit/05b854c168ffc29e7164e6bc764bb05aa5f00660). So logging is working from the start and thread may be initialized later. 

So your change is good because it clear things but if it is actually fixing something then i would like to understand what. Could that be that in your case backend activated was initialized before log_core?

**Comment by mike-scott** [comment]
@nordic-krch @pfalcon I will re-word the title / description to use language like:
"log processing" and "log buffering" to clarify the intended effect of this patch.

The only change this patch makes is when the "log processing" starts if CONFIG_LOG_PROCESS_THREAD is enabled.   This allows for the default behavior to process logs for drivers and other kernel processes prior to the main() function call.

**Review by nordic-krch — APPROVED** [review]

**Comment by pfalcon** [comment]
@mike-scott : Thanks!

File: subsys/logging/log_core.c [hunk 1]
```diff
@@ -570,16 +571,23 @@ static void log_process_thread_func(void *dummy1, void *dummy2, void *dummy3)
...
+
 static int enable_logger(struct device *arg)
 {
 	ARG_UNUSED(arg);
+#ifdef CONFIG_LOG_PROCESS_THREAD
+	/* start logging thread */
+	k_thread_create(&logging_thread, logging_stack,
+			K_THREAD_STACK_SIZEOF(logging_stack),
+			log_process_thread_func, NULL, NULL, NULL,
+			CONFIG_LOG_PROCESS_THREAD_PRIO, 0, K_NO_WAIT);
```
Comment (nashif) [inline subsys/logging/log_core.c]: please set the name of the thread. Previously it was set in the K_THREAD_DEFINE, when creating threads with k_thread_create this needs to be set with k_thread_name_set.

**Review by nashif — DISMISSED** [review]

**Comment by mike-scott** [comment]
Changes:
- Set the thread name
- Updated the patch description for clarification

**Comment by mike-scott** [comment]
@pfalcon @nordic-krch does the new patch description clarify the intent of this patch?

**Comment by mike-scott** [comment]
@nashif gentle ping regarding your review

**Comment by mike-scott** [comment]
@nashif should I change assignee to someone else?

This change is needed before I remove the asynch startup of wnc-m14a2a modem which would fix start up for samples without needing to wait for interface events

**Comment by nordic-krch** [comment]
@nashif, ping.

**Review by pfalcon — DISMISSED** [review]
As @nordic-krch confirmed this fixes my issue https://github.com/zephyrproject-rtos/zephyr/issues/11008, let me approve this too.

**Comment by jukkar** [comment]
Without this ethernet debugging does not work, this is very much needed.

**Comment by jukkar** [comment]
> The fix is very much needed

Sorry, I dismissed wrong review

**Comment by pfalcon** [comment]
If I happened to give -1 instead of intended +1, d'oh, sorry!

**Comment by jukkar** [comment]
> If I happened to give -1 instead of intended +1, d'oh, sorry!

No, it was my mistake. I accidentally dismissed your +1.

**Pull request merged**